### DISCUSSION TITLE
Data Labeling for PyTorch Geometric

### DISCUSSION THREAD

**Discussion by bensch98** [pyg-team/pytorch_geometric#4745] [Q&A]
Hi,
I just came across PyTorch Geometric. I'm very familiar with PyTorch in general, especially CNNs, but I am wondering if Graph Neural Networks can fit my needs better for my current project. I have questions regarding these two topics:

1. I want to perform semantic segmentation on 3D meshes (.stl, .ply, .obj files) to determine the vectors to certain parts/regions (subsets of the whole mesh) of CAD models (more specifically identifying cylindrical holes and planar surfaces with a rail). I think in this case it's actually called part segmentation. In general are GNNs more suitable for this task than 3D-CNNs?
2. Most papers apparently don't talk about how its data was labeled, hence I couldn't find any labeling tool for my use case. Does anyone know good labeling tools for 3D meshes? What is the best file format to use? Is there a dataset you can recommend I should get started with? I've been looking into the ABC dataset and MPI-FAUST dataset. Are there any best practices or examples how I should structure my data during labeling.

I appreciate any recommendations (books, papers, repositories, ...). I'm currently already going through the AMMI Geometric Deep Learning Course (2021) on YouTube and Graph Neural Networks book by Lingfei Wu, Peng Cui, Jian Pei and Liang Zhao.

**Comment by rusty1s** [comment]
1. GNNs have the advantage that they are actually able to operate on the raw input representation, and not on some reduced representations like 3D-CNNs do. 3D-CNNs also have the disadvantage to require dense representations, which will result in high computation costs once your number of voxel grows. As such, numerous other approaches have been proposed to overcome the limitations of 3D-CNNs - one of them being GNNs operating on point clouds or mesh like data. Other approaches use, e.g., SDF representations of surfaces/models (`DeepSDF`).
2. I cannot give any recommendation here, I am sorry.
